src/Tuura/Concept/Concat/Abstract.hs

```diff
@@ -1,51 +1,57 @@
 module Tuura.Concept.Concat.Abstract (
     Interface (..),
+    InitialValue (..),
     Concept (..),
     initialConcept, arcConcept,
     interfaceConcept,
-    (.&&.), (.||.),
     ) where
 
-import Control.Applicative
+import Data.Monoid
 
 -- Abstract concepts
 -- * s is the type of states
 -- * e is the type of events
 
 data Interface = Unused | Input | Internal | Output deriving (Ord, Eq, Show)
 
+data InitialValue = Undefined | Defined { getDefined :: Bool } | Inconsistent deriving (Eq, Show)
+
+instance Monoid InitialValue where
+    mempty = Undefined
+
+    mappend Inconsistent _ = Inconsistent
+    mappend _ Inconsistent = Inconsistent
+    mappend Undefined x = x
+    mappend x Undefined = x
+    mappend (Defined x) (Defined y) = if x == y then Defined x else Inconsistent
+
+
 data Concept s e a = Concept
                    {
-                       initial :: s -> Bool,
+                       initial :: a -> InitialValue,
                        arcs    :: [(e, e)],
                        interface :: a -> Interface
                    }
 
 instance Monoid (Concept s e a) where
     mempty = Concept
              {
-                 initial = const True,
+                 initial = const mempty,
                  arcs    = [],
                  interface = const Unused
              }
     mappend a b = Concept
                   {
-                      initial = initial a .&&. initial b,
+                      initial = \s -> initial a s <> initial b s,
```

Comment:
I think it may be possible to simplify this to just `initial = initial a <> initial b` because there is a standard instance generation rule: `Monoid a => Monoid (b -> a)`, i.e. if we know how to compose values of type `a`, we also know how to compose functions returning type `a`.